### PR TITLE
Fetch calls

### PR DESCRIPTION
**Pull request by colev1**
-Created fetch calls to import datasets for countries and continents.
-Set state on App with three properties; countryData, continentData, and countryInput so these 
-Implemented a method on app to find the specific country in the country dataset.
-Passed props from Search to be states on App.js

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -21,22 +28,47 @@ class App extends Component {
           })
         })
         .catch(error => console.log(error))
-        console.log(this.state.countryData)
+
+        console.log('fetched')
+    }
```
Comment (Dhanciles) [inline src/App.js]: Great work on these methods! This allows us to have our data on page load, as we are calling both methods to grap our data in our 'componentDidMount()' methods.

File: src/App.js [hunk 2]
```diff
@@ -21,22 +28,47 @@ class App extends Component {
...
+        console.log(chosenCountryObj);
     }
 
   render() {
     return (
       <div className="App">
         <header className="App-header">
-          <Search />
+          <Search countryData={this.state.countryData} continentData={this.state.continentData} updateCountryInput={this.updateCountryInput} findCountry={this.findCountry}/>
           <Filter />
```
Comment (Dhanciles) [inline src/App.js]: Passing down the state of our app as props for our search component.

File: src/Search.js [hunk 3]
```diff
@@ -10,20 +10,26 @@ class Search extends Component {
...
     }
 
     takeCountry = (event, country) => {
       event.preventDefault();  
-      document.querySelector('.input').value = '';
+      this.props.updateCountryInput(this.state.countryName)
+      this.props.findCountry(this.state.countryName);
+      this.setState({
+        countryName: ''
+      })
```
Comment (Dhanciles) [inline src/Search.js]: Passing up our data that we received in our search components state into the 'findCountry()' method that was passed down as props. Now our method is being invoked to find the country the user has searched for.

File: src/App.js [hunk 4]
```diff
@@ -21,22 +28,47 @@ class App extends Component {
...
+    }
+
+    findCountry = (countryName) => {
+      let chosenCountryObj = this.state.countryData.countries.find((country) => {
+        if(country.name.toLowerCase() === countryName) {
+          return country;
+        }
+      })
+        console.log(chosenCountryObj);
     }
```
[the remark this answers is not in the document]
Comment (colev1) [reply src/App.js]: Okay!

**Pull request merged**